File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_6_crypto_tracker/after/tests/PortfolioTests.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.after.tests;

import com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.after.core.domain.PortfolioManager;    
import com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.after.infrastructure.adapters.FakePriceProvider;

/**
 * ARCHITECTURAL TEST
 */
public class PortfolioTests {

    private PortfolioTests() {
        // Private constructor to hide the implicit public one
    }

    public static void run() throws Exception {
        System.out.println("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---");
        
        // Arrange
        FakePriceProvider fakeAdapter = new FakePriceProvider(50000.0);
        PortfolioManager manager = new PortfolioManager(fakeAdapter);

        // Act
        System.out.println("Test Action: Calculating value of 2 BTC at fixed $50,000 price...");
        double value = manager.calculateTotalValue(2.0);

        // Assert
        if (value == 100000.0) {
            System.out.println("SUCCESS: The portfolio correctly calculated $100,000. Test is stable!");
        } else {
            System.out.println("FAIL: Math error in Core logic.");
        }
    }
}

File: Chapter 05/C#/Section_5_6_Crypto_Tracker/After/Tests/PortfolioTests.cs
using System;
using System.Threading.Tasks;
using Chapter05.CryptoTracker.After.Core.Domain;
using Chapter05.CryptoTracker.After.Infrastructure.Adapters;

namespace Chapter05.CryptoTracker.After.Tests
{
    /// <summary>
    /// ARCHITECTURAL TEST
    /// Fulfills the Scribe role by proving the test passes with 100% reliability.
    /// </summary>
    public static class PortfolioTests
    {
        public static async Task Run()
        {
            Console.WriteLine("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---");
            
            // Arrange
            // We use the Fake adapter so we know EXACTLY what the price is ($50,000)
            var fakeAdapter = new FakePriceProvider(50_000m);
            var manager = new PortfolioManager(fakeAdapter);

            // Act
            Console.WriteLine("Test Action: Calculating value of 2 BTC at fixed $50,000 price...");
            var value = await manager.CalculateTotalValue(2m);

            // Assert
            if (value == 100_000m) 
            {
                Console.WriteLine("SUCCESS: The portfolio correctly calculated $100,000. Test is 100% stable!");
            }
            else 
            {
                Console.WriteLine("FAIL: Math error in Core logic.");
            }
        }
    }
}